### ISSUE TITLE
option to load in CBVs from data directory

### ISSUE THREAD

**Issue by benjaminpope** [lightkurve/lightkurve#1107]
#### Problem description
CBVCorrector currently loads MAST CBVs by default, or else you have to set up design matrices yourself. But I want to run CBVCorrector on an HPC system without access to the internet.

#### Expected behavior
I have already downloaded all TESS CBVs to a directory on the cluster. It would be nice to just pass it a "cbv_dir='my_cbv_dir'" option and it will look for them in there, but otherwise display the same behaviour.

**Comment by orionlee** [comment]
In the interim, there appears to have an (ugly) workaround. You'd need to write some logic to load the local CBVs though.

```python
    corrector = CBVCorrector(lc, do_not_load_cbvs=True)
    corrector.cbvs = custom_cbv_loading_from_local_dir(lc)
```

See: 
http://docs.lightkurve.org/reference/api/lightkurve.correctors.CBVCorrector.html#lightkurve.correctors.CBVCorrector.__init__

**Comment by barbierimauro** [comment]
I have a similar issue of BenjaminPope, but related to lightcurves.
I have all the lightcurves downloaded in a local archive, and I have also the list of the filenames for each TICID.
There is a way to tell to lightkurve to open all the local fits files of one TICID ? (I mean creating a unique object)

**Comment by jcsmithhere** [comment]
Hi @benjaminpope and @orionlee, 

Thank you for the feedback. This is a great recommendation! I see a way I can add an option to pass a path to load the CBVs without much work at all. But in the mean time, yes, you can pass a custom design matrix and still use the fitting algorithms in the CBVCorrector class. However, I think I can add the local path option in the near future.

**Comment by benjaminpope** [comment]
Thanks! Yeah I think the main issue is that there is currently a very nice API for automatically selecting which CBVs to use. It would be nice to be able to use the *same code* inside an HPC core, pointing to a pre-saved directory of CBVs, as you would in a notebook.

**Comment by orionlee** [comment]
@jcsmithhere An option to pass a path to load the CBVs would also be helpful in software maintenance. 

Right now, in the continuous integration tests, the CBV codes are only tested on python 3.8+Linux environment (as all tests rely on external data do). With a local path option, some CBV tests can be added to the regular test suite that would in turn be tested in a variety of platform/Python.

**Comment by orionlee** [comment]
@barbierimauro if you have the list of local paths of the FITS file of a given TIC, you can read each of them using [`read` function](https://docs.lightkurve.org/reference/api/lightkurve.LightCurve.read.html):


```python
import lightkurve as lk
lc = lk.read(path_of_a_fits_file)
```

**Comment by barbierimauro** [comment]
@orionlee I am actually using this method when the TICID is observed in only 1 sector and it works fine.
I am wondering how I can do it if there are multiple sectors for the same TICID ?
assuming that this is my situation : /some_path/sect01/tic_number.fits /some_path/sect02/tic_number.fits etc
how I can import all of them in the same lc object ?

**Comment by orionlee** [comment]
@barbierimauro In this case, you could `stitch` multiple `lc` objects together as one, each loaded from the FITS file of the underlying TIC - sector.

```python
import lightkurve as lk

# path_list_of_a_tic is the list of file path of a given TIC's fits file
lc = lk.LightCurveCollection([lk.read(p) for p in path_list_of_a_tic]).stitch()
    
```

The function is defined in: 
https://github.com/lightkurve/lightkurve/blob/81bc8697480afd371e01a9c526487f25e414ba7d/src/lightkurve/collections.py#L174-L176

See also: https://docs.lightkurve.org/tutorials/2-creating-light-curves/2-1-combining-multiple-quarters.html

**Comment by orionlee** [comment]
@barentsen API doc for `LightCurveCollection`, etc. is missing on http://docs.lightkurve.org/ . It was fixed and merged a while back, but the changes have not yet been uploaded to docs.lightkurve.org.

**Comment by barbierimauro** [comment]
@orionlee many thanks for the explanation, I think that this solve my issue.

**Comment by orionlee** [comment]
@benjaminpope For your use case of running CBVCorrector on an HPC system without access to the internet, even if the CBVs can be loaded locally, wouldn't there be a problem during the actual `cbvcorrector.correct()` call, when it searches and downloads for neighboring lightcurves during optimization?

**Comment by benjaminpope** [comment]
Yes well - tackling one part of the problem at the time. For these bright stars the functionality of using nearby stars for optimization is useless, and we will need custom functionality for this too.
